packages/editor/src/store/selectors.js

```diff
@@ -1995,6 +1995,17 @@ export function isPostLocked( state ) {
 	return state.postLock.isLocked;
 }
 
+/**
+ * Returns whether post saving is locked.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} Is locked.
+ */
+export function isPostSavingLocked( state ) {
+	return find( state.postSavingLock, ( lock ) => lock ) ? true : false;
```

Comment:
My hunch upon seeing this is: Why can't this just be a condition of `Object.keys( state.postSavingLock ).length > 0` ? Why do we hang on to locks in state after the unlock occurs?

Which would require further revision to the reducer to use [`omit`](https://lodash.com/docs/4.17.10#omit):

```js
case 'LOCK_POST_SAVING':
	return { ...state, [ action.lockName ]: true };

case 'UNLOCK_POST_SAVING':
	return omit( state, action.lockName );
```

Reply:
yea, that bothered me as well, good suggest - i will refine that

Reply:
Adjusted in https://github.com/WordPress/gutenberg/pull/10649/commits/949ed50506190992c40c175891bc3a397cd6c3ba. Any reason to use `Object.keys( state.postSavingLock ).length > 0` instead of `state.postSavingLock.length > 0`?

Reply:
@adamsilverstein there is a very good reason: Object.length is undefined. 🙂 Opened a pull request to fix this little glitch https://github.com/WordPress/gutenberg/pull/11636